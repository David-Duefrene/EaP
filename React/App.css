.App {
	text-align: center;
	background-color: #282c34;
	min-height: 99vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	color: white;
}

button {
	font-size: calc(10px + 2vmin);
}
