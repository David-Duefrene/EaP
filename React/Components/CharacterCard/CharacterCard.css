.Card {
	display: flex;
	margin: 1%;
	background-color: var(--main-color);
	color: white;
	border-radius: 10px;
	box-shadow: 0px 0px 7px var(--accent-color);
	transition: filter 1s;
}

.Card:hover {
	filter: sepia(60%);
}

.Portrait {
	max-width: 100%;
	max-height: 100%;
	border-radius: 10px;
}

.CardBody {
	list-style: none;
}
