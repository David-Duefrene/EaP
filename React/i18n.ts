import i18n from 'i18next'
import { initReactI18next } from 'react-i18next'

import keys from './public/locales/en/keys.json'
import enums from './public/locales/en/enums.json'

i18n.use(initReactI18next).init({
	lng: 'en',
	resources: {
		en: { keys, enums },
	},
	interpolation: {
		escapeValue: false,
	},
})

export default i18n
