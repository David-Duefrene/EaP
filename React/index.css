html {
	--main-color: #282c34;
	--main-off-color: white;
	--accent-color: #61dafb;
	--highlight-color: #1f8ead;
}

body {
	margin: 0;
	font-family: -apple-system, BlinkMacSystemFont, 'Segoe UI', 'Roboto', 'Oxygen',
		'Ubuntu', 'Cantarell', 'Fira Sans', 'Droid Sans', 'Helvetica Neue',
		sans-serif;
	-webkit-font-smoothing: antialiased;
	text-align: center;
	background-color: var(--main-color);
	min-height: 99vh;
	display: flex;
	flex-direction: column;
	align-items: center;
	justify-content: center;
	color: var(--main-off-color);
	overflow: hidden;
}

#root {
	width: 100vw;
	height: 100vh;
}

button, .Button {
	font-size: calc(10px + 2vmin);
	display: inline-block;
	background-image: radial-gradient(var(--highlight-color), var(--main-color));
	padding: 0.5em 1em;
	color: var(--main-off-color);
	border: 1px solid var(--main-off-color);
	border-radius: 5px;
	padding: 5px;
	margin: 5px;
	cursor: pointer;
	transition: all .5s;
	height: 1em;
}

button:hover, button:focus, .Active, .Button:hover, .Button:focus {
	filter: invert(90%);
}

button:active {
	background-color: var(--main-color);
	color: var(--accent-color);
	transition: all 0.1s;
}

code {
  font-family: source-code-pro, Menlo, Monaco, Consolas, 'Courier New',
    monospace;
}
