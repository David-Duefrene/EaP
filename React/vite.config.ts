import { fileURLToPath } from 'url'
import { defineConfig } from 'vite'
import react from '@vitejs/plugin-react'

// https://vitejs.dev/config/
export default defineConfig({
	plugins: [ react() ],
	base: './',
	build: {
		outDir: '../Electron/vite-build',
	},
})
