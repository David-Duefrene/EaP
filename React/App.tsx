import React, { useState, useEffect } from 'react'

const { PrismaClient } = require('@prisma/client')

import AddCharacter from './AddCharacter/AddCharacter'
import logo from './logo.svg'
import './App.css'

const prisma = new PrismaClient()

const App = () => {
	const [ characters, setCharacters ] = useState([])

	useEffect(() => {
		prisma.character.findMany().then((d) => setCharacters(d))
	}, [])

	return (
		<div className='App'>
			<header className='App-header'>
				<img src={logo} className='App-logo' alt='logo' />
				<p>Hello Vite, React, GraphQL, and Electron!</p>
				<p>env is {process.env.NODE_ENV}</p>
				<AddCharacter />
				{characters.map((el, key) => {
					return (
						<div key={key}>
							<div>{el.name}</div>
							<div>{el.characterID}</div>
						</div>
					)
				}) }
			</header>
		</div>
	)
}

export default App
