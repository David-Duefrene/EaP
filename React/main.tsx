import React from 'react'
import ReactDOM from 'react-dom/client'
import {
	Route, Routes, BrowserRouter,
} from 'react-router-dom'

import Home from './Pages/Home/Home'
// import Blueprints from './Pages/Character/Blueprints/Blueprints'
// import ContactNotifications from './Pages/Character/ContactNotifications/ContactNotifications'
// import CorpHistory from './Pages/Character/CorpHistory/CorpHistory'
// import CorpRoles from './Pages/Character/CorpRoles/CorpRoles'
// import MedalList from './Pages/Character/Medals/MedalList'
// import Notifications from './Pages/Character/Notifications/Notifications'
// import Standings from './Pages/Character/Standings/Standings'
import Titles from './Pages/Character/Titles/Titles'
import ErrorPage from './Pages/ErrorPage/ErrorPage'

import NavBar from './Components/NavBars/MainNavBar'
import Notification from './Components/Buttons/Notification/Notification'
import './index.css'

import CharacterQuery from '../Types/APIResponses/PrismaQueries/Character/CharacterSheetQueries.type'
import Title from '../Types/APIResponses/EveOfficial/Title.types'

interface FindAll {
	characters: () => Promise<CharacterQuery[]>,
	characterSheets: () => Promise<CharacterQuery[]>,
}

interface Auth {
	login: () => void,
}

interface GetCharacter {
	titles: (characterID: bigint) => Promise<Title>,
}

declare global {
	interface Window {
		findAll: FindAll
		auth: Auth
		getCharacter: GetCharacter
	}
  }

const routes =
	<Routes>
		<Route path='/' element={<Home />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID' element={<Titles />} errorElement={<ErrorPage />} />
		{/* <Route path='character/:characterID/blueprints' element={<Blueprints />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID/contact-notification' element={<ContactNotifications />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID/corp-history' element={<CorpHistory />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID/corp-roles' element={<CorpRoles />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID/medal-list' element={<MedalList />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID/notifications' element={<Notifications />} errorElement={<ErrorPage />} />
		<Route path='character/:characterID/standings' element={<Standings />} errorElement={<ErrorPage />} /> */}
		<Route path='character/:characterID/Titles' element={<Titles />} errorElement={<ErrorPage />} />
	</Routes>

ReactDOM.createRoot(document.getElementById('root')!).render(
	<React.StrictMode>
		<BrowserRouter>
			<NavBar />
			{routes}
			<Notification />
		</BrowserRouter>
	</React.StrictMode>,
)
