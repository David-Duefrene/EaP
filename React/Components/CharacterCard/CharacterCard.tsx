import React from 'react'
import { Link } from 'react-router-dom'

import CharacterQuery from '../../../Types/APIResponses/PrismaQueries/Character/CharacterSheetQueries.type'
import './CharacterCard.css'

const CharacterCard = (props: { character: CharacterQuery }) => {
	const {
		characterID, name, allianceID, birthday, bloodlineID, corporationID, gender, raceID, securityStatus,
	} = props.character

	return (
		<Link key={characterID} className='Card' to={`character/${characterID}`}>
			<img alt={`${name}'s character portrait`} className='Portrait' src={`https://images.evetech.net/characters/${characterID}/portrait?tenant=tranquility&size=256`} />
			<ul className='CardBody'>
				<li>{`Name: ${name}`}</li>
				<li>{`Character ID: ${characterID}`}</li>
				<li>{`Alliance: ${allianceID}`}</li>
				<li>{`Birthday: ${birthday.toLocaleString()}`}</li>
				<li>{`Bloodline: ${bloodlineID}`}</li>
				<li>{`Corporation: ${corporationID}`}</li>
				<li>{`Gender: ${gender}`}</li>
				<li>{`Race: ${raceID}`}</li>
				<li>{`Security Status: ${securityStatus.toFixed(2)}`}</li>
			</ul>
		</Link>
	)
}

export default CharacterCard
