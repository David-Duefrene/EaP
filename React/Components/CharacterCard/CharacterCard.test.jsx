// @vitest-environment jsdom
import React from 'react'
import {
	createBrowserRouter, createRoutesFromElements, RouterProvider, Route,
} from 'react-router-dom'

import '../../i18n'

import {
	expect, test, afterEach, describe, beforeEach,
} from 'vitest'
import {
	cleanup, render, screen,
} from '@testing-library/react'

import CharacterCard from './CharacterCard'

describe('CharacterCard', () => {
	const character = {
		characterID: 123,
		name: 'Test Name',
		alliance: 456,
		birthday: '2021-01-01',
		bloodlineName: 'Jin-Mei',
		corporation: 901,
		description: 'Test Description',
		gender: 'male',
		raceName: 'Gallente',
		securityStatus: 10,
	}

	beforeEach(() => {
		/*
		 * This is a workaround for the following issue:
		 * Link needs to be rendered inside a Router
		 * // TODO: Remove this workaround when link is refactored into the home page
		 */
		const router = createBrowserRouter(createRoutesFromElements(
			<Route path='/' element={<CharacterCard character={character} />} />,
		))
		render(<RouterProvider router={router}></RouterProvider>)
	})

	afterEach(cleanup)

	test('should render the character name', () => {
		expect(screen.getByText(`Name: ${character.name}`)).toBeTruthy()
	})

	test('should render the character ID', () => {
		expect(screen.getByText(`Character ID: ${character.characterID}`)).toBeTruthy()
	})

	test('should render the character Alliance ID', () => {
		expect(screen.getByText(`Alliance: ${character.alliance}`)).toBeTruthy()
	})

	test('should render the character birthday', () => {
		expect(screen.getByText(`Birthday: ${character.birthday}`)).toBeTruthy()
	})

	test('should render the character Bloodline ID', () => {
		expect(screen.getByText(`Bloodline: ${character.bloodlineName}`)).toBeTruthy()
	})

	test('should render the character Corp ID', () => {
		expect(screen.getByText(`Corporation: ${character.corporation}`)).toBeTruthy()
	})

	test('should render the character Gender', () => {
		expect(screen.getByText('Gender: Male')).toBeTruthy()
	})

	test('should render the character Race', () => {
		expect(screen.getByText(`Race: ${character.raceName}`)).toBeTruthy()
	})

	test('should render the character Security Status', () => {
		expect(screen.getByText(`Security Status: ${character.securityStatus.toFixed(2)}`)).toBeTruthy()
	})
})
