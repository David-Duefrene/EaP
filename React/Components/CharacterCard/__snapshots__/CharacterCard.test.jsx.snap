// Vitest Snapshot v1

exports[`CharacterCard > should render the character card 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="Card"
      >
        <img
          alt="Test Name's character portrait"
          class="Portrait"
          src="https://images.evetech.net/characters/123/portrait?tenant=tranquility&size=256"
        />
        <ul
          class="CardBody"
        >
          <li>
            Test Name
          </li>
          <li>
            Character ID: 
            123
          </li>
          <li>
            Alliance ID: 
            456
          </li>
          <li>
            Character birthday: 
            2021-01-01
          </li>
          <li>
            Bloodline ID: 
            789
          </li>
          <li>
            Corp ID: 
            901
          </li>
          <li>
            Gender: 
            male
          </li>
          <li>
            Race ID: 
            234
          </li>
          <li>
            Security Status: 
            10
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="Card"
    >
      <img
        alt="Test Name's character portrait"
        class="Portrait"
        src="https://images.evetech.net/characters/123/portrait?tenant=tranquility&size=256"
      />
      <ul
        class="CardBody"
      >
        <li>
          Test Name
        </li>
        <li>
          Character ID: 
          123
        </li>
        <li>
          Alliance ID: 
          456
        </li>
        <li>
          Character birthday: 
          2021-01-01
        </li>
        <li>
          Bloodline ID: 
          789
        </li>
        <li>
          Corp ID: 
          901
        </li>
        <li>
          Gender: 
          male
        </li>
        <li>
          Race ID: 
          234
        </li>
        <li>
          Security Status: 
          10
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
